avro_schema/frontend.lua

```diff
@@ -665,7 +665,15 @@ copy_data = function(schema, data, visited)
         end
         return data
     elseif schematype == 'double' or schematype == 'float' then
-        return 0 + tonumber(data)
+        local xtype = type(data)
+        if xtype == "number" then
+            return data
+        else
+            if xtype == "cdata" then
```

Comment:
Do you want to check for our implementation of number64? Is it worth to check stricter?

```
type(data) == 'cdata' and tostring(ffi.typeof(data)) == 'ctype<uint64_t>'
```

Reply:
1. There is a bunch of possible types, e.g. `1LL` is `ctype<int64_t>`, char, short, long, float...
2. I think that rely on `tonumber` function is ok, which mean that we should only check that `tonumber` result is not `nil`

Reply:
Good point.